msteams-platform/bots/bot-concepts.md

```diff
@@ -0,0 +1,592 @@
+    await next();
+});
+
+```
+
+---
+
+The following code provides an example of bot activity for a one-to-one chat:
+
+# [C#](#tab/dotnet)
+
+* [SDK reference](/dotnet/api/microsoft.bot.schema.activityextensions.removerecipientmention?view=botbuilder-dotnet-stable&preserve-view=true)
+
+* [Sample code reference](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/app-hello-world/csharp/Microsoft.Teams.Samples.HelloWorld.Web/Bots/MessageExtension.cs#L19)
+
+```csharp
+
+// Handle message activity
+protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+{
+    // Remove recipient mention text from Text property.
+    // Use with caution because this function is altering the text on the Activity.
+    turnContext.Activity.RemoveRecipientMention();
+    var text = turnContext.Activity.Text.Trim().ToLower();
+
+    // Sends a message activity to the sender of the incoming activity.
+    await turnContext.SendActivityAsync(MessageFactory.Text($"Your message is {text}."), cancellationToken);
+}
+```
+
+# [Node.js](#tab/nodejs)
+
+* [SDK reference](/javascript/api/botbuilder-core/turncontext?view=botbuilder-ts-latest&preserve-view=true#botbuilder-core-turncontext-sendactivity)
+* [Sample code reference](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-receive-channel-messages-withRSC/nodejs/server/bot/botActivityHandler.js#L20)
+
+```javascript
+this.onMessage(async (context, next) => {
+    // MessageFactory.text(): Specifies the type of text data in a message attachment.
+    await context.sendActivity(MessageFactory.text("Your message is: " + context.activity.text));
+    await next();
+});
+```
+
+---
+
+### Bot logic
+
+Bot logic incorporates the fundamental rules and decision-making frameworks that dictate a bot's actions and interactions. It outlines how the bot interprets user input, formulates responses, and participates in conversations.
+
+In Teams, the bot logic processes incoming activities from one or more of your bot channels and in response generates outgoing activities. It's still true of bots derived from the Teams activity handler class, which first checks for Teams activities. After checking for Teams activities, it passes all other activities to the Bot Framework's activity handler.
+
+# [C#](#tab/csharp)
+
+#### Core Bot Framework handlers
+
+>[!NOTE]
+>
+>* Except for the **added** and **removed** members' activities, all the activity handlers described in this section continue to work as they do with a non-Teams bot.
+>* `onInstallationUpdateActivityAsync()` method is used to get Teams Locale while adding the bot to Teams.
+
+Activity handlers are different in context of a team, where a new member is added to the team instead of a message thread.
+
+The list of handlers defined in `ActivityHandler` includes the following events:
+
+| Event | Handler | Description |
```

Comment:
I would suggest to add like Handler/SDK Method and also for handler names we can add example
Actual: OnTurnAsync

Proposed: OnTurnAsync()

Reply:
@ChetanSharma-msft I've updated as suggested.